pysd/py_backend/xmile/SMILE2Py.py

```diff
@@ -153,17 +160,27 @@ def __init__(self, model_namespace):
 
         self.grammar = parsimonious.Grammar(grammar)
 
-    def parse(self, text, context='eqn'):
+    def parse(self, text, element, context='eqn'):
         """
            context : <string> 'eqn', 'defn'
                 If context is set to equation, lone identifiers will be parsed as calls to elements
                 If context is set to definition, lone identifiers will be cleaned and returned.
         """
-        # !TODO! Should remove the inline comments from `text` before parsing the grammar
+        
+        # Remove the inline comments from `text` before parsing the grammar
         # http://docs.oasis-open.org/xmile/xmile/v1.0/csprd01/xmile-v1.0-csprd01.html#_Toc398039973
+        text = re.sub(r"\{[^}]*\}", "", text)
+        
         self.ast = self.grammar.parse(text)
         self.context = context
-        return self.visit(self.ast)
+        self.element = element
+        self.new_structure = []
+        
+        py_expr = self.visit(self.ast)
+        
+        return ({
+            'py_expr': py_expr
+        }, self.new_structure)
 
     def visit_conditional_statement(self, n, vc):
         _IF, _1, condition_expr, _2, _THEN, _3, then_expr, _4, _ELSE, _5, else_expr = vc
```

Comment:
I like this. I was wondering about using python's `if... then.. else` syntax, so that you can short-circuit the evaluation of outputs, but this is clean and consistent with what we have elsewhere. I think it's good.